pkg/stanza/operator/input/file/input.go

```diff
@@ -37,20 +38,47 @@ func (i *Input) Stop() error {
 	return i.fileConsumer.Stop()
 }
 
-func (i *Input) emit(ctx context.Context, token emit.Token) error {
-	if len(token.Body) == 0 {
-		return nil
+func (i *Input) emitBatch(ctx context.Context, tokens []emit.Token) error {
```

Comment:
Given that this is in the hot path, we might be better off flattening these functions. It seems like a lot of extra error checking and wrapping, where we really just need to put a loop around what was already there, switch from return to append err + continue.

Reply:
I have inlined the `convertToken` function in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36663/commits/de66483ba1d472dc40775856e92fecf46ab5c89e. Should I inline the `convertTokens` function as well?